File: ruby/lib/mudbase/qr_token.rb
# frozen_string_literal: true

require "securerandom"

module Mudbase
  # A random, unguessable single-use check-in code, mirroring the reference web app's
  # `generateQrToken()` (`crypto.randomUUID().replace(/-/g, "")`, 32 hex chars / 122 bits of
  # entropy). `SecureRandom.hex(16)` yields the same 32-hex-character shape at 128 bits - not a
  # security credential in the cryptographic sense (this is a demo ticketing app, not a payments
  # system), just collision-resistant enough for a QR check-in token.
  module QrToken
    def self.generate
      SecureRandom.hex(16)
    end
  end
end

File: php/src/Support/QrToken.php
<?php

declare(strict_types=1);

namespace App\Support;

/**
 * A random, unguessable single-use check-in code. Not a security credential in the cryptographic
 * sense (this is a demo ticketing app, not a payments system) - collision resistance against
 * 128 bits of `random_bytes` entropy is more than sufficient for a QR check-in token. Mirrors the
 * reference app's `generateQrToken()` (src/lib/utils.ts, `crypto.randomUUID().replace(/-/g, "")`)
 * closely enough for this demo: a 32-hex-character opaque token with comparable entropy.
 */
final class QrToken
{
    public static function generate(): string
    {
        return bin2hex(random_bytes(16));
    }
}
